api/src/server/event.rs

```diff
@@ -0,0 +1,331 @@
+    }
+    let event_bytes = get_block(&event_cid, &car_blocks, store).await?;
+    let event: unvalidated::Event =
+        serde_ipld_dagcbor::from_slice(&event_bytes).context("decoding event")?;
+    let init_id = match event {
+        unvalidated::Event::Signed(event) => {
+            let link = event
+                .link()
+                .ok_or_else(|| anyhow!("event should have a link"))?;
+
+            let payload_bytes = get_block(&link, &car_blocks, store).await?;
+            let payload: unvalidated::Payload<Ipld> =
+                serde_ipld_dagcbor::from_slice(&payload_bytes).context("decoding payload")?;
+            match payload {
+                unvalidated::Payload::Init(_) => event_cid,
+                unvalidated::Payload::Data(payload) => payload.id(),
+            }
+        }
+        unvalidated::Event::Time(event) => event.id(),
+    };
+
+    let init_bytes = get_block(&init_id, &car_blocks, store).await?;
+    let init_event: unvalidated::Event =
+        serde_ipld_dagcbor::from_slice(&init_bytes).context("decoding init event")?;
+    if let unvalidated::Event::Signed(event) = init_event {
+        let link = event
+            .link()
+            .ok_or_else(|| anyhow!("init event should have a link"))?;
+
+        let payload_bytes = get_block(&link, &car_blocks, store).await?;
+        let payload: unvalidated::Payload<Ipld> =
+            serde_ipld_dagcbor::from_slice(&payload_bytes).context("decoding init payload")?;
+        if let unvalidated::Payload::Init(payload) = payload {
+            Ok(EventId::new(
+                &network,
+                payload.header().sep(),
+                payload.header().model().as_slice(),
+                payload
+                    .header()
+                    .controllers()
+                    .first()
+                    .ok_or_else(|| anyhow!("init header should contain at least one controller"))?,
+                &init_id,
+                &event_cid,
+            ))
+        } else {
+            bail!("init event payload is not well formed")
+        }
+    } else {
+        bail!("init event should be a signed event")
+    }
+}
+
+// Helper function to get the block first from the CAR file data or otherwise from the store.
+async fn get_block<S: AccessModelStore>(
+    cid: &Cid,
+    car_blocks: &HashMap<Cid, Vec<u8>>,
+    store: &S,
+) -> Result<Vec<u8>> {
+    if let Some(bytes) = car_blocks.get(cid) {
+        Ok(bytes.clone())
+    } else {
+        Ok(store
+            .get_block(cid)
+            .await?
+            .ok_or_else(|| anyhow!("cannot find block {cid}"))?)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use std::str::FromStr;
+
+    use super::*;
+
+    use async_trait::async_trait;
+    use expect_test::{expect, Expect};
+    use mockall::{mock, predicate};
+    use tracing_test::traced_test;
+
+    const INIT_EVENT_CID: &str = "bagcqcerar2aga7747dm6fota3iipogz4q55gkaamcx2weebs6emvtvie2oha";
```

Comment:
can you include a reference to how these things were generated?  Ideally a script that's checked in to the repo?  I hate having tests with hard-coded data that no one knows how to recreate in case anything changes and the structures need to be updated.

Reply:
@stephhuynh18 Can you give me some insight on how these were created? My assumption was that you started up js-ceramic created a stream and gave it an update and waited for it be anchored and then grabbed the CAR data from the logs? 

@stbrody Agreed we need a way to generate this data. I don't think we have a simple way to do so now. As such I was most interested in getting real world data so we can have more confidence that the code works as intended.